src/azul/uuids.py

```diff
@@ -0,0 +1,77 @@
+import uuid
+
+
+class InvalidUUIDError(Exception):
+
+    def __init__(self, entity_id: str):
+        super().__init__(f'{entity_id} is not a valid uuid.')
+
+
+class InvalidUUIDPrefixError(Exception):
+
+    def __init__(self, prefix: str):
+        super().__init__(f'{prefix} is not a valid uuid prefix.')
+
+
+def validate_uuid(uuid_str: str, allow_uppercase: bool = False) -> uuid.UUID:
```

Comment:
```suggestion
def validate_uuid(uuid_str: str) -> uuid.UUID:
```

Unless we actually need this.